### ISSUE TITLE
When opening a local file using drag and drop, it cannot update/save changes

### ISSUE THREAD

**Issue by mawid6** [jgraph/drawio#2864]
### Preflight Checklist

* [x] I agree to follow the [Code of Conduct](https://github.com/jgraph/drawio/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Describe the bug**
When opening a local file from device **_using the menus_**, things work as expected, doing a save (ctrl-s) will update the opened file.
When opening the same file **_by dragging_** it into the drawio editor, the behavior is different: Saving does not work, and doing a save (ctrl-s) will not do a save and update the opened file - instead it opens the general save modal dialogue (**_update:_** or - as of today Aug 15th 2022 - it downloads a new copy of the file).

**To Reproduce**
1. Start drawio/diagrams.net web app
2. Locate a diagram on your local drive. Drag that file onto the drawio canvas to open it.
3. File opens. Optional: Make a small change.
4. Press **ctrl-s to save your changes** to the diagram file.

**Expected behavior**
A save is completed - the local file is updated (a dialogue to allow the browser app access to the file system is displayed, you click ok and after that the changes are committed to disk).

**Actual behavior**
The general save modal dialogue is displayed, offering the choice of storage.
**_Update:_** or - as of today Aug 15th 2022 - it downloads a new copy of the file.

**draw.io version (In the Help->About menu of the draw.io editor):**
- draw.io version 18.0.7

**Desktop (please complete the following information):**
 - OS: windows 10 
 - Browser chrome
 - Version  101.0.4951.67 (Official Build) (64-bit)

**Additional information**
In the same theme of issues related to opening/saving diagrams as #2825 ( https://github.com/jgraph/drawio/issues/2825 ).

**Comment by mawid6** [comment]
In the same theme of issues related to opening/saving diagrams as #2825.

**Comment by ashrafteleb** [comment]
Dragging a file to the editor is **importing** the file in the current one. It is not equivalent to opening.

**Comment by mawid6** [comment]
> Dragging a file to the editor is **importing** the file in the current one. It is not equivalent to opening.

**_TLDR:_** No, that is not accurate. That is _one_ of the drag operations, dragging to the _canvas_. I not talking about dragging to the canvas to import. I am speaking of dragging to the app, to open. Can I please ask you to check and reopen this ticket?

----- 
**_Longer:_** 
Oh, ok, so you are saying it is by design - that there is no support for opening a file via drag and drop. 

#### Wrong. I am pretty sure your statement is incorrect. Why:

(A) **Docs.** Reading the [documentation on diagrams.net in the official FAQ](https://drawio.freshdesk.com/support/solutions/articles/16000067784-open-an-existing-diagram-file-in-diagrams-net), where @davidjgraph writes:

> Drag and drop: Alternatively, there is a faster way to open a diagram stored on your local device - simply drag the diagram file onto the diagrams.net drawing canvas.

tells me different.

(B) **Both import & open are supported, as different operations:** The behavior of the webapp UI closely mimics the desktop app, and they both provide different feedback to distinguish between **two different drag and drop targets**, providing different cues to each. 

If you drag to the:

* **canvas**: the contents of the dragged diagram are indeed **_imported_** into the canvas. The cue is that the _canvas_ is outlined with a dashed box. It is a smaller/inner outline. 
* **window**: the diagram is **opened**. This applies to both in both webapp and desktop app. Again, the cue is - this time - that the entire window (and not just the canvas) is outlined with a dashed box. It is a larger/outer outline.


Again, in this issue, I am talking about opening via drag, which you can do by dragging to the window - not the canvas.

Now, this ticket actually talks about a problem that happens **_after_ the diagram has opened**: In the desktop app, everything works as you would expect, and the open operation has achieved the same result as if you open via the menu. However, in the webapp, as described in this ticket, each time I press ctrl-s to save, the original file is not updated as would be expected.



So I would be grateful if you would reopen this issue, since I spent some time describing it.

**Comment by mawid6** [comment]
( Also note: I have updated that with my brief testing today, the save dialogue is not opened, but rather, a copy of the updated diagram is immediately downloaded. 
It may be that both behaviors appear, depending on something - I am not sure. It does not matter much - it does not alleviate or worsen the issue described. )

**Comment by davidjgraph** [comment]
I do not write the article, it just shows who created the page.

The article is wrong, I've deleted it.

**Comment by mawid6** [comment]
OK @davidjgraph, I did not know that, seems like a fair assumption given that it had your name on it - but I understand now, thanks for clarifying.

Thanks for reopening this! 
I think it would be truly great to have this working.

**Comment by ukrolelo** [comment]
+1 crucial  for me moving  tech department away from MS visio. It would be awesome if it would work like this as a logic dictates that if i would like to save it in different location i would use save as. But if i drag and drop above the canvas (window) it would open the file i dragged in.

**Comment by davidjgraph** [comment]
@ukrolelo does the visio web application correctly obtain the file system handle to the file when you drag and drop the file into the visio editor in a browser and so save back to the file?

**Comment by ukrolelo** [comment]
@davidjgraph we are still using ms visio 2010 its win32 app... so looking forward to move away from old OS and from MS.... i think we might start from drawio desktop (It is possible to open the file from file explorer) , but the problem is that after time we would be forced to update to another future windows to keep using newest drawio as newest drawio is from last year not compatible with windows 7/8. So i wanted something that would be compatible on any OS (through web browser). And picking from open file menu is a dealbreaker...

**Comment by ukrolelo** [comment]
P.S. why i also really want to move to drawio as by reading the drawio file with external script i can gather (text) material list in the drawing. It is very interesting! :)

**Comment by davidjgraph** [comment]
The problem is comparing a desktop application to one in a web browser isn't like for like. The File API in a browser is nothing like the file handle functionality that a desktop app has available.

So, unless someone can show me that this is technically possible in some way, the conclusion is that it's not.

**Comment by ukrolelo** [comment]
Got it, i was thinking that it's possible as when you Open from Device and open the local file,i am be able to save it to the same location if i press save. So if i understood correctly that D&D a file is different than File -> Open from Device ?
